packages/app/src/components/PageEditor/EmojiAutoCompleteHelper.js

```diff
@@ -1,149 +1,88 @@
-  //     else if ((data.keywords != null) && data.keywords.find((elem) => { return elem.indexOf(term) > -1 })) {
-  //       results4.push(data.shortname);
-  //     }
-  //   }
-
-  //   let results = results1.concat(results2).concat(results3).concat(results4);
-  //   results = results.slice(0, maxLength);
-
-  //   return results;
-  // }
+  constructor() {
+    this.showHint = this.showHint.bind(this);
+  }
+
+
+  /**
+   * try to find emoji terms and show hint
+   * @param {any} editor An editor instance of CodeMirror
+   */
+  showHint(editor) {
+    // see https://regex101.com/r/gy3i03/1
+    const pattern = /:[^:\s]+/;
+
+    const currentPos = editor.getCursor();
+    // find previous ':shortname'
+    const sc = editor.getSearchCursor(pattern, currentPos, { multiline: false });
+    if (sc.findPrevious()) {
+      const isInputtingEmoji = (currentPos.line === sc.to().line && currentPos.ch === sc.to().ch);
+      // return if it isn't inputting emoji
+      if (!isInputtingEmoji) {
+        return;
+      }
+    }
+    else {
+      return;
+    }
+
+    /*
+     * https://github.com/weseek/growi/issues/703 is caused
+     * because 'editor.display.viewOffset' is zero
+     *
+     * call stack:
+     *   1. https://github.com/codemirror/CodeMirror/blob/5.42.0/addon/hint/show-hint.js#L220
+     *   2. https://github.com/codemirror/CodeMirror/blob/5.42.0/src/edit/methods.js#L189
+     *   3. https://github.com/codemirror/CodeMirror/blob/5.42.0/src/measurement/position_measurement.js#L372
+     *   4. https://github.com/codemirror/CodeMirror/blob/5.42.0/src/measurement/position_measurement.js#L315
+     */
+    UpdateDisplayUtil.forceUpdateViewOffset(editor);
+
+    // see https://codemirror.net/doc/manual.html#addon_show-hint
+    editor.showHint({
+      completeSingle: false,
+      // closeOnUnfocus: false,  // for debug
+      hint: () => {
+        const matched = editor.getDoc().getRange(sc.from(), sc.to());
+        const term = matched.replace(':', ''); // remove ':' in the head
+
+        // get a list of shortnames
+        const emojiIndex = new NimbleEmojiIndex(emojiData);
+        const shortnames = emojiIndex.search(term);
+        if (shortnames.length >= 1) {
+          return {
+            list: this.generateEmojiRenderer(shortnames),
+            from: sc.from(),
+            to: sc.to(),
+          };
+        }
+      },
+    });
+  }
+
+  /**
+   * see https://codemirror.net/doc/manual.html#addon_show-hint
+   * @param {string[]} emojiShortnames a list of shortname
+   */
+  // TODO 13 Jan 2022
+  generateEmojiRenderer(emojiShortnames) {
+    return emojiShortnames.map((emoji) => {
+      return {
+        text: emoji.colons,
+        className: 'crowi-emoji-autocomplete',
+        render: (element) => {
+          element.innerHTML = `<div class="img-container">${emoji.native}</div>`
+            + `<span class="shortname-container">${emoji.colons}</span>`;
+        },
+      };
+    });
+  }
```

Comment:
these code are not needed.

please just show emoji-mart UI with React hook
> In case of that a word :peac is added on editor, emoji-mart will be appeared like the following image.
<img width="711" alt="Screen Shot 2022-01-13 at 20 47 32" src="https://user-images.githubusercontent.com/59536731/149324877-c30a5573-5cd1-4a67-ab8b-cb05b7c1b47e.png">